projects/canviewer/canviewer.py

```diff
@@ -0,0 +1,163 @@
+from canserver import init_can
+from ui import render_dashboard
+
+import argparse
+from os import listdir
+import time
+import signal
+import sys
+import numpy as np
+
+from rich.layout import Layout
+from rich.live import Live
+
+db = None
+
+FILTERS = [
+    {
+        "can_id": 0x000,
+        "can_mask": 0x7E0,
+        "extended": False,
+    },
+    {
+        "can_id": 0x0A0,
+        "can_mask": 0x7FF,
+        "extended": False,
+    },
+]
+
+SHUTDOWN_NODES = {
+    "ss_bspd": None,  # BSPD
+    "ss_is": None,  # THROTTLE
+    "ss_bots": None,  # THROTTLE
+    "ss_estop_driver": None,  # THROTTLE
+    "ss_hvd": None,  # AIR
+    "ss_hvd_conn": None,  # AIR
+    "ss_mpc": None,  # AIR
+    "ss_bms": None,  # AIR
+    "ss_imd": None,  # AIR
+    "ss_tsms": None,  # AIR
+}
+
+VEHICLE_VALUES = {}
+
+VEHICLE_STATES = {
+    "AIR Control": [None, None],
+    "BMS": [None, None],
+    "Throttle": [None, None],
+}
+
+
+def sigint(signal_received, frame):
+    exit(0)
+
+
+def convertVtoT(x, Vin=3, R1=10000, R2=100000, T2=348.15, beta=3988):
+    """
+    converts voltage drop data to temperature
+    x: array containing only the voltages
+    """
+    Vout = float(x)
+    thermistor_R1 = (Vout * R1) / (Vin - Vout)
+    temperature = 1 / ((np.log(thermistor_R1 / R2) / beta) + (1 / T2)) - 273.15
+
+    return temperature
+
+
+def get_val(signal, data):
+    val = data.get(signal)
+
+    if val != None:
+        return str(val)
+    else:
+        return None
+
+
+def get_ss_val(signal, data):
+    val = data.get(signal)
+
+    if val:
+        if val == "OPEN":
+            return "[red]OPEN"
+        elif val == "CLOSED":
+            return "[green]CLOSED"
+        else:
+            return None
+
+
+def rx_callback(msg, db):
+    global SHUTDOWN_NODES
+    global VEHICLE_VALUES
+
+    try:
+        message = db.decode_message(msg.arbitration_id, msg.data)
+    except Exception as e:
```

Comment:
Do we know what exception we're catchign here? I trust that this works but it would be cleaner if we could specify that